nemo/lightning/pytorch/strategies.py

```diff
@@ -491,6 +505,22 @@ def training_step(self, dataloader_iter, *args: Any, **kwargs: Any) -> STEP_OUTP
 
             return out
 
+    @override
+    def optimizer_step(
+        self,
+        optimizer: torch.optim.Optimizer,
+        closure: Callable[[], Any],
+        model: Optional[Union["pl.LightningModule", nn.Module]] = None,
+        **kwargs: Any,
+    ) -> Any:
+        optimizer_output = super().optimizer_step(optimizer, closure, model, **kwargs)
+
+        if isinstance(optimizer, McoreDistributedOptimizer):
```

Comment:
Currently will only work for MCore Distributed Optimizers. I'm open to making this more general by returning a dictionary from `McoreDistributedOptimizer.step()`. I don't know if PTL does anything with the outputs of the optimizer step.

Reply:
@marcromeyn ^^ for the design choice